docs/quickstart.rst

```diff
@@ -0,0 +1,553 @@
+
+In this example ``'test_auth_provider'`` is a dummy local authentication provider, it's linked to the ``'test_identity_provider'`` as specified in ``MULTIPASS_PROVIDER_MAP``. Specifying ``'type'`` as ``'static'`` links those providers to our ``StaticAuthProvider`` and ``StaticIdentityProvider`` example classes(More on those classes later).
+
+In ``identities`` settings of ``'test_auth_provider'`` we specify key-value pairs of username (Pig) and password (pig123), those are used for authentication by Multipass. In this example, the usernames are used as unique identifier for users. In ``identities`` settings of ``'test_identity_provider'`` we assign info keys dictionary to usernames. We also need to specify these keys in ``MULTIPASS_IDENTITY_INFO_KEYS``.
+
+.. code-block:: python
+
+	MULTIPASS_AUTH_PROVIDERS = {
+	    'test_auth_provider': {
+		'type': 'static',
+		'title': 'Insecure dummy auth',
+		'identities': {
+		    'Pig': 'pig123',
+		    'Bunny': 'bunny123'
+		}
+	    }
+	}
+
+	MULTIPASS_IDENTITY_PROVIDERS = {
+	    'test_identity_provider': {
+		'type': 'static',
+		'identities': {
+		    'Pig': {'email': 'guinea.pig@example.com', 'name': 'Guinea Pig', 'affiliation': 'Pig University'},
+		    'Bunny': {'email': 'bugs.bunny@example.com', 'name': 'Bugs Bunny', 'affiliation': 'Bunny Inc.'}
+		},
+		'groups': {
+		    'Admins': ['Pig'],
+		    'Everybody': ['Pig', 'Bunny'],
+		}
+	    }
+	}
+
+	MULTIPASS_PROVIDER_MAP = {
+	    'test_auth_provider': 'test_identity_provider'
+	}
+	
+	MULTIPASS_IDENTITY_INFO_KEYS = ['email', 'name', 'affiliation']
+
+We also need to specify the template with a login form for our provider:
+
+.. code-block:: python
+
+	MULTIPASS_LOGIN_FORM_TEMPLATE = 'login_form.html'
+	
+Implementing providers
+~~~~~~~~~~~~~~~~
+
+Let's create our authentication provider class, which should inherit from ``AuthProvider``.
+We should also specify the login form class (which inherits from ``FlaskForm``) which we use for login of this provider.
+
+.. code-block:: python
+
+	class StaticLoginForm(FlaskForm):
+	    username = StringField('Username', [DataRequired()])
+	    password = PasswordField('Password', [DataRequired()])
+
+	class StaticAuthProvider(AuthProvider):
+	    login_form = StaticLoginForm
+    
+.. _local_login:
+
+Local provider login
+~~~~~~~~~~~~~~~~
+
+The process of handling URLs for login is the same as with external providers therefore please check the :ref:`external_login` part of this guide.
+
+The only difference is that  ``multipass.process_login`` calls method ``_login_form`` which renders a template specified in ``MULTIPASS_LOGIN_FORM_TEMPLATE`` with the ``login_form`` specified in the authentication provider class.
+
+Once the form is submitted method ``process_local_login`` of authentication provider class is called. In this method you have to implement your authentication logic. 
```

Comment:
"Once the form is submitted *comma* **the** method... of **the** authentication provider class..."